gprofiler/profilers/java.py

```diff
@@ -259,11 +288,71 @@ def make_application_metadata(self, process: Process) -> Dict[str, Any]:
         # that loads other libs.
         libjvm_elfid = get_mapped_dso_elf_id(process, "/libjvm")
 
-        metadata = {"java_version": version, "libjvm_elfid": libjvm_elfid}
+        metadata = {
+            "java_version": version,
+            "libjvm_elfid": libjvm_elfid,
+            "jvm_flags": self.get_jvm_flags_serialized(process),
```

Comment:
Please collect this in try/except to refrain from distrupting the collection of other metadata items (such as `java_version`). Log upon error and use `None` for `jvm_flags`.

Reply:
why we don't try except also for ```get_java_version```?

Reply:
It's a good point, we should. You can fix it in this PR like I did [here](https://github.com/Granulate/gprofiler/pull/656/files), or keep for a later PR. For new features that we add - we add them more robust to begin with

Reply:
So there's no point in catching exceptions and returning null in inner methods.

Reply:
Wdym?

Reply:
Ah with respect to this method. Yes, you are right. Letting the exception raise, catching it here and marking `error:` is preferred.

In a separate PR we can fix `java_version` and `libjvm_elfid` to also not fail us